Pip wrong dependencies package

Hello,

TL;DR : satosa_oidcop should update package requirements to use idpyoidc>=3.0.0

OS info:
NAME="CentOS Stream"
VERSION="9"

pip 21.3.1 from /usr/lib/python3.9/site-packages/pip (python 3.9)
Python 3.9.18

When trying to use satosa with satosa-oidcop front-end will result in a fail attempt with the following message in debug logs.
```
[2024-07-02 11:19:19,330] [ERROR] [satosa.proxy_server.make_app] Failed to create WSGI app.
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/satosa/proxy_server.py", line 192, in make_app
    res1 = WsgiApplication(satosa_config)
  File "/usr/local/lib/python3.9/site-packages/satosa/proxy_server.py", line 118, in __init__
    super().__init__(config)
  File "/usr/local/lib/python3.9/site-packages/satosa/base.py", line 45, in __init__
    frontends = load_frontends(self.config, self._auth_req_callback_func,
  File "/usr/local/lib/python3.9/site-packages/satosa/plugin_loader.py", line 68, in load_frontends
    frontend_modules = _load_plugins(config.get("CUSTOM_PLUGIN_MODULE_PATHS"), config["FRONTEND_MODULES"],
  File "/usr/local/lib/python3.9/site-packages/satosa/plugin_loader.py", line 181, in _load_plugins
    instance = module_class(callback, internal_attributes, module_config, base_url,
  File "/usr/local/lib/python3.9/site-packages/satosa_oidcop/idpy_oidcop.py", line 591, in __init__
    self.app = oidcop_app(conf)
  File "/usr/local/lib/python3.9/site-packages/satosa_oidcop/core/application.py", line 23, in oidcop_application
    app = oidc_provider_init_app(config)
  File "/usr/local/lib/python3.9/site-packages/satosa_oidcop/core/application.py", line 16, in oidc_provider_init_app
    app.server = Server(config, cwd=folder)
  File "/usr/local/lib/python3.9/site-packages/idpyoidc/server/__init__.py", line 72, in __init__
    self.context = EndpointContext(
  File "/usr/local/lib/python3.9/site-packages/idpyoidc/server/endpoint_context.py", line 240, in __init__
    _supports = self.supports()
  File "/usr/local/lib/python3.9/site-packages/idpyoidc/server/endpoint_context.py", line 406, in supports
    res.update(endpoint.supports())
  File "/usr/local/lib/python3.9/site-packages/idpyoidc/server/endpoint.py", line 485, in supports
    res[key] = val()
  File "/usr/local/lib/python3.9/site-packages/idpyoidc/claims.py", line 248, in get_signing_algs
    return sorted(_list, key=cmp_to_key(alg_cmp))
  File "/usr/local/lib/python3.9/site-packages/idpyoidc/claims.py", line 233, in alg_cmp
    _pos1 = SIGNING_ALGORITHM_SORT_ORDER.index(a[0:2])
ValueError: 'Ed' is not in list
```
After digging, in package cryptojwt/jws/jws.py :
```
SIGNER_ALGS = {
    "HS256": HMACSigner("SHA256"),
    "HS384": HMACSigner("SHA384"),
    "HS512": HMACSigner("SHA512"),
    "RS256": RSASigner("RS256"),
    "RS384": RSASigner("RS384"),
    "RS512": RSASigner("RS512"),
    "ES256": ECDSASigner("ES256"),
    "ES256K": ECDSASigner("ES256K"),
    "ES384": ECDSASigner("ES384"),
    "ES512": ECDSASigner("ES512"),
    "PS256": PSSSigner("SHA256"),
    "PS384": PSSSigner("SHA384"),
    "PS512": PSSSigner("SHA512"),
    "EdDSA": EDDSASigner(),
    "Ed25519": EDDSASigner("Ed25519"),
    "Ed448": EDDSASigner("Ed448"),
    "none": None,
}
```
Which led to idpyoidc/server/construct.py :
```
ALG_SORT_ORDER = {"RS": 0, "ES": 1, "HS": 2, "PS": 3, "no": 4}
[...]
def assign_algorithms(typ):
    if typ == "signing_alg":
        # Pick supported signing algorithms from crypto library
        # Sort order RS, ES, HS, PS                                                 <-- is not aware of Ed signer from cryptojwt
        sign_algs = list(SIGNER_ALGS.keys())
        return sorted(sign_algs, key=cmp_to_key(sort_sign_alg))
    elif typ == "encryption_alg":
        return jwe.SUPPORTED["alg"]
    elif typ == "encryption_enc":
        return jwe.SUPPORTED["enc"]
``` 
Which led to ValueError: 'Ed' is not in list when starting satosa

Forcing the installation of idpyoidc v3.0.0 with pip fix the issue seamlessly.
```pip install --force-reinstall -v "idpyoidc==3.0.0"```

Source : https://github.com/IdentityPython/idpy-oidc/commit/7129c23c62ab0b20336c05be8c74d0449d97da18

--- a/setup.py
+++ b/setup.py
@@ -38,6 +38,6 @@ setup(
     install_requires=[
         "satosa>=8.0.0",
         "pymongo>=3.11,<5.0",
-        "idpyoidc>=2.1.0,<2.2.0",
+        "idpyoidc>=4.1.0,<4.2.0",
     ],
 )

--- a/tests/test_oidcop.py
+++ b/tests/test_oidcop.py
@@ -583,7 +583,7 @@ class TestOidcOpFrontend(object):
         _access_token = _token_resp['access_token']
         context.request_authorization = f"{_token_resp['token_type']} {_access_token[:-2]}"
         userinfo_resp = frontend.userinfo_endpoint(context)
-        assert userinfo_resp.status == "403"
+        assert userinfo_resp.status == "403 Forbidden"
         
         # Test UserInfo endpoint
         context.request = {}
@@ -683,7 +683,7 @@ class TestOidcOpFrontend(object):
         basic_auth = urlsafe_b64encode(credentials.encode("utf-8")).decode("utf-8")
         context.request_authorization = f"Basic {basic_auth}"
         token_resp = frontend.token_endpoint(context)
-        assert token_resp.status == '403'
+        assert token_resp.status == '403 Forbidden'
         assert json.loads(token_resp.message)['error'] == 'invalid_request'
 
     def test_load_cdb_basicauth(self, context, frontend):
@@ -807,7 +807,7 @@ class TestOidcOpFrontend(object):
         # Test FAULTY refresh_token
         context.request["refresh_token"] = _res.get('refresh_token')[:-2]
         refresh_resp = frontend.token_endpoint(context)
-        assert refresh_resp.status == "403"
+        assert refresh_resp.status == "403 Forbidden"
         _res = json.loads(refresh_resp.message)
         assert _res['error'] == "invalid_grant"
         assert _res['error_description'] == "Invalid refresh token"

